lib/Makefile

```diff
@@ -292,16 +295,20 @@ libzstd : $(LIBZSTD)
 
 .PHONY: lib
 lib : libzstd.a libzstd
+	@if [ $(ZSTD_NO_MULTITHREAD) = 1 ]; then \
+		echo "multithreading disabled"; \
+	else \
+		echo "multithreading enabled"; \
+	fi
+
+# multithreading enabled by default unless otherwise specified
+ifneq ($(ZSTD_NO_MULTITHREAD), 1)
+  CPPFLAGS += -DZSTD_MULTITHREAD
+  LDFLAGS  += -pthread
+endif
 
-
-# note : do not define lib-mt or lib-release as .PHONY
+# note : do not define lib-release as .PHONY
 # make does not consider implicit pattern rule for .PHONY target
-
-%-mt : CPPFLAGS += -DZSTD_MULTITHREAD
```

Comment:
Minor : modified scope

The purpose of `%-mt` rule is to convert any base rule (`lib`, `libzstd`, `libzstd.a`) into a multithreaded variant. Heck, even `make all` could become `make all-mt`.

By removing these lines, we make invoking these targets an error.
I get it that they become redundant,
but _maybe_ there are users who are using them right now
and it would be great that their build script still work.

Suggestion : keep it, as a simple redirector, so that invoking them still works.
`%-mt : %`
Maybe update the message to indicate that `-mt` suffix is no longer necessary, and that the target will be removed in a future version (v1.6.0?).